### PR TITLE
Feature/dsnd 468 company metrics api initialize spring boot

### PR DESCRIPTION
**Pull request by sampipparampil**
Initialised Spring boot application for company-metrics-api.
feature-dsnd-468.

### CODE REVIEW

File: docker-compose.yml.v0.10 [hunk 1]
```diff
@@ -0,0 +1,17 @@
+version: '3'
+services:
```
Comment (zahidm3) [inline docker-compose.yml.v0.10:2]: We can delete this file as API components dont need Kafka v0.10 on docker compose

File: src/itest/java/uk/gov/companieshouse/company/metrics/controller/HealthCheckControllerITest.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@AutoConfigureMockMvc
+@SpringBootTest
+public class HealthCheckControllerITest {
+
+    @Autowired
+    private MockMvc mockMvc;
+
```
Comment (zahidm3) [inline src/itest/java/uk/gov/companieshouse/company/metrics/controller/HealthCheckControllerITest.java:19]: We can also this class, once Cucumber BDD tests introduced we can run integration tests without mocks

**Review by zahidm3 — CHANGES_REQUESTED** [review]

File: src/itest/java/uk/gov/companieshouse/company/metrics/controller/HealthCheckControllerITest.java (hunk 2 above)
Comment (sampipparampil) [reply src/itest/java/uk/gov/companieshouse/company/metrics/controller/HealthCheckControllerITest.java]: Removed HealthCheckControllerITest.java

**Comment by waqasahmed29** [comment]
@sampipparampil just add one more file please > DS_Store, please check charges-data-api repo for this file too.

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+.PHONY: clean
+clean:
+	@# Help: Reset repo to pre-build state (i.e. a clean checkout state)
+	mvn clean 
+
+.PHONY: build
+build:
+	@# Help: Pull down any dependencies and compile code into an executable if required
+	mvn package -Dmaven.test.skip=true
+	cp ./target/$(artifact_name)-$(version).jar ./target/$(artifact_name).jar
```
[the remark this answers is not in the document]
Comment (sampipparampil) [reply Makefile]: DSND:468: Updated Makefile and added version file

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+lint: lint/docker-compose sonar
+	@# Help: Run all lint/* targets and sonar
+
+.PHONY: lint/docker-compose
+lint/docker-compose:
+	@# Help: Lint docker file
+	docker-compose -f docker-compose.yml config
+
+
+
```
[the remark this answers is not in the document]
Comment (sampipparampil) [reply Makefile:100]: DSND:468: added start.sh file

**Review by zahidm3 — APPROVED** [review]

**Pull request merged**